app/ui-react/packages/auto-form/src/widgets/FormArrayComponent.tsx

```diff
@@ -11,120 +11,119 @@ export interface IFormArrayComponentProps extends IFormArrayControlProps {
-                          value: rowValue[property.name],
-                        })
-                      )}
-                      <div
-                        key={controlGroupName}
-                        {...formGroupAttributes}
-                        {...arrayControlAttributes}
-                      >
-                        <label
-                          htmlFor={toValidHtmlId(`${controlGroupName}-control`)}
-                          className="control-label"
-                          {...controlLabelAttributes}
-                        />
-                        <div id={toValidHtmlId(`${controlGroupName}-control`)}>
-                          <div className="form-array-control__array-controls">
-                            {options.showSortControls && (
-                              <>
-                                <TextButton
-                                  onClick={() => {
-                                    this.props.move(index, index - 1);
-                                  }}
-                                  enable={index > 0}
-                                >
-                                  <i className="fa fa-arrow-circle-o-up" />
-                                </TextButton>
-                                <TextButton
-                                  onClick={() => {
-                                    this.props.move(index, index + 1);
-                                  }}
-                                  enable={index < values.length - 1}
-                                >
-                                  <i className="fa fa-arrow-circle-o-down" />
-                                </TextButton>
-                              </>
-                            )}
-                            <TextButton
-                              onClick={() => this.props.remove(index)}
-                              enable={values.length > minElements}
-                            >
-                              <i className="fa fa-trash-o" />
-                            </TextButton>
-                          </div>
+      <div className="alert alert-warning">
+        <span>No Array definition supplied for array type</span>
+      </div>
+    );
+  }
+  const definition = props.property.arrayDefinition;
+  const options =
+    props.property.arrayDefinitionOptions ||
+    ({} as IFormArrayDefinitionOptions);
+  const formGroupAttributes = options.formGroupAttributes || {};
+  const fieldAttributes = options.fieldAttributes || {};
+  const controlLabelAttributes = options.controlLabelAttributes || {};
+  const arrayControlAttributes = options.arrayControlAttributes || {};
+  const arrayRowTitleAttributes = options.arrayRowTitleAttributes || {};
+  const minElements = options.minElements || 0;
+  const values =
+    props.form.values[props.name] || props.property.defaultValue || [];
+  const myId = toValidHtmlId(props.name);
+  return (
+    <div id={myId} data-testid={myId}>
```

Comment:
For a following PR, this JSX could be made more readable by taking out the inner bits of the loops in stand-alone components.